### PR TITLE
Rename Tools to Components in devfile

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Component is a better name. It's more generic and tools is ok for IDE tools (language suppport, debugging etc...) but less for IDE runtimes (application servers, databases etc....) or editors (e.g. theia).

And we need to do the renaming also to align with odo terminology

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/12836

#### Release Notes
Change in devfile structure -- rename toolc to components

#### Docs PR
N/A

### CODE REVIEW

**Comment by mshaposhnik** [comment]
ci-test

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/dockerimage/DockerimageComponentToWorkspaceApplier.java [hunk 1]
```diff
@@ -43,46 +43,46 @@
...
 
   /**
-   * Applies changes on workspace config according to the specified dockerimage tool.
+   * Applies changes on workspace config according to the specified dockerimage component.
    *
    * @param workspaceConfig workspace config on which changes should be applied
-   * @param dockerimageTool dockerimage tool that should be applied
-   * @param contentProvider optional content provider that may be used for external tool resource
-   *     fetching
+   * @param dockerimageTool dockerimage component that should be applied
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/dockerimage/DockerimageComponentToWorkspaceApplier.java]: parameter name should be changed as well

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/kubernetes/KubernetesComponentToWorkspaceApplier.java [hunk 2]
```diff
@@ -286,18 +299,18 @@ private void estimateCommandsMachineName(
...
 
     String machineName = Names.machineName(pod, pod.getSpec().getContainers().get(0));
-    toolCommands.forEach(c -> c.getAttributes().put(MACHINE_NAME_ATTRIBUTE, machineName));
+    componentCommands.forEach(c -> c.getAttributes().put(MACHINE_NAME_ATTRIBUTE, machineName));
   }
 
-  private List<HasMetadata> unmarshalToolObjects(Tool k8sTool, String toolLocalContent)
-      throws DevfileRecipeFormatException {
+  private List<HasMetadata> unmarshalComponentObjects(
+      Component k8sComponent, String toolLocalContent) throws DevfileRecipeFormatException {
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/kubernetes/KubernetesComponentToWorkspaceApplier.java]: `toolLocalContent` -> `componentLocalContent`?

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/dockerimage/DockerimageComponentToWorkspaceApplier.java [hunk 3]
```diff
@@ -43,46 +43,46 @@
...
    */
   @Override
   public void apply(
       WorkspaceConfigImpl workspaceConfig,
-      Tool dockerimageTool,
+      Component dockerimageTool,
       FileContentProvider contentProvider)
       throws DevfileException {
     checkArgument(workspaceConfig != null, "Workspace config must not be null");
     checkArgument(dockerimageTool != null, "Tool must not be null");
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/dockerimage/DockerimageComponentToWorkspaceApplier.java]: `"Component must not be null"`

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/kubernetes/KubernetesComponentToWorkspaceApplier.java [hunk 4]
```diff
@@ -310,13 +323,14 @@ private void estimateCommandsMachineName(
     }
   }
 
-  private String asYaml(Tool tool, List<HasMetadata> list) throws DevfileRecipeFormatException {
+  private String asYaml(Component tool, List<HasMetadata> list)
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/kubernetes/KubernetesComponentToWorkspaceApplier.java]: `Component tool` -> `Component component`

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/editor/EditorComponentToWorkspaceApplier.java [hunk 5]
```diff
@@ -9,61 +9,65 @@
...
   @Override
   public void apply(
-      WorkspaceConfigImpl workspaceConfig, Tool editorTool, FileContentProvider contentProvider)
-      throws DevfileException {
+      WorkspaceConfigImpl workspaceConfig,
+      Component editorComponent,
+      FileContentProvider contentProvider) {
     checkArgument(workspaceConfig != null, "Workspace config must not be null");
-    checkArgument(editorTool != null, "Tool must not be null");
+    checkArgument(editorComponent != null, "Tool must not be null");
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/editor/EditorComponentToWorkspaceApplier.java]: `Component must not be null`

File: wsmaster/che-core-api-devfile/src/test/java/org/eclipse/che/api/devfile/server/convert/component/kubernetes/KubernetesComponentToWorkspaceApplierTest.java [hunk 6]
```diff
@@ -66,38 +66,41 @@
...
     workspaceConfig = new WorkspaceConfigImpl();
   }
 
   @Test(
       expectedExceptions = DevfileException.class,
       expectedExceptionsMessageRegExp =
-          "Fetching content of file `local.yaml` specified in `local` field of tool `foo` is not "
+          "Fetching content of file `local.yaml` specified in `local` field of component `foo` is not "
               + "supported. Please provide its content in `localContent` field. Cause: fetch is not supported")
   public void shouldThrowExceptionWhenRecipeToolIsPresentAndContentProviderDoesNotSupportFetching()
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/test/java/org/eclipse/che/api/devfile/server/convert/component/kubernetes/KubernetesComponentToWorkspaceApplierTest.java]: I think methods should be renamed as well

**Comment by sleshchenko** [comment]
I think it is better to mention that this PR already contains some part of needed Docs changes instead of `N/A`. 
Do you plan to update diagrams in https://github.com/redhat-developer/devfile ?

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/plugin/PluginProvisioner.java [hunk 7]
```diff
@@ -33,13 +33,13 @@
  *
  * @author Sergii Leshchenko
  */
-public class PluginProvisioner implements ToolProvisioner {
+public class PluginProvisioner implements ComponentProvisioner {
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/plugin/PluginProvisioner.java:36]: fix java doc
```
Provision chePlugin tools 
```

**Review by sleshchenko — APPROVED** [review]
LGTM

**Review by skabashnyuk — APPROVED** [review]

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1649/) doesn't show any regression against this Pull Request.

**Pull request merged**